src/ragged/_spec_set_functions.py

```diff
@@ -47,8 +52,31 @@ def unique_all(x: array, /) -> tuple[array, array, array, array]:
     https://data-apis.org/array-api/latest/API_specification/generated/array_api.unique_all.html
     """
 
-    x  # noqa: B018, pylint: disable=W0104
-    raise NotImplementedError("TODO 128")  # noqa: EM101
+    if isinstance(x, ragged.array):
+        if len(x) == 1:
+            return unique_all_result(
+                values=ragged.array(x),
+                indices=ragged.array([0]),
+                inverse_indices=ragged.array([0]),
+                counts=ragged.array([1]),
+            )
+        else:
```

Comment:
If `x == 1`, does the `else` clause of this `if`-`else` produce the correct result? If so, simplify the code and just use that clause.

```suggestion
```

(Future enhancements will be easier if there are fewer `if`-`else` statements. In particular, if we ever want to put `ragged.array` into Dask, expressions that check `len(x)` wouldn't be possible because Dask prepares a set of instructions for other computers to run, and on those other computers, the length may be zero or not zero; hence, it's something we can't check on the first computer.)